core/auth/src/main/java/software/amazon/awssdk/auth/signer/AwsS3V4Signer.java

```diff
@@ -15,209 +15,18 @@
 
 package software.amazon.awssdk.auth.signer;
 
-import static software.amazon.awssdk.auth.signer.internal.Aws4SignerUtils.calculateRequestContentLength;
-import static software.amazon.awssdk.auth.signer.internal.SignerConstant.X_AMZ_CONTENT_SHA256;
-
-import java.io.InputStream;
-import java.util.Optional;
 import software.amazon.awssdk.annotations.SdkPublicApi;
-import software.amazon.awssdk.auth.credentials.CredentialUtils;
-import software.amazon.awssdk.auth.signer.internal.AbstractAws4Signer;
-import software.amazon.awssdk.auth.signer.internal.Aws4SignerRequestParams;
-import software.amazon.awssdk.auth.signer.internal.chunkedencoding.AwsChunkedEncodingConfig;
-import software.amazon.awssdk.auth.signer.internal.chunkedencoding.AwsChunkedEncodingInputStream;
-import software.amazon.awssdk.auth.signer.internal.chunkedencoding.AwsS3V4ChunkSigner;
-import software.amazon.awssdk.auth.signer.params.Aws4PresignerParams;
-import software.amazon.awssdk.auth.signer.params.AwsS3V4SignerParams;
-import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
-import software.amazon.awssdk.http.ContentStreamProvider;
-import software.amazon.awssdk.http.SdkHttpFullRequest;
-import software.amazon.awssdk.utils.BinaryUtils;
+import software.amazon.awssdk.auth.signer.internal.AbstractAwsS3V4Signer;
 
 /**
  * AWS4 signer implementation for AWS S3
  */
 @SdkPublicApi
-public final class AwsS3V4Signer extends AbstractAws4Signer<AwsS3V4SignerParams, Aws4PresignerParams> {
-
-    private static final String CONTENT_SHA_256 = "STREAMING-AWS4-HMAC-SHA256-PAYLOAD";
-
-    /**
-     * Sent to S3 in lieu of a payload hash when unsigned payloads are enabled
-     */
-    private static final String UNSIGNED_PAYLOAD = "UNSIGNED-PAYLOAD";
-    private static final String CONTENT_LENGTH = "Content-Length";
-
+public final class AwsS3V4Signer extends AbstractAwsS3V4Signer {
```

Comment:
Moved the implementation to an abstract class to be able to override methods on the signer for testing